src/OrchardCore.Modules/OrchardCore.Contents/AdminMenu.cs

```diff
@@ -55,6 +58,10 @@ public async Task BuildNavigationAsync(string name, NavigationBuilder builder)
                         var ci = await _contentManager.NewAsync(contentTypeDefinition.Name);
                         var cim = await _contentManager.PopulateAspectAsync<ContentItemMetadata>(ci);
                         var createRouteValues = cim.CreateRouteValues;
+
+                        //We add a returnUrl to the contents list for these menu items
+                        createRouteValues.Add("returnUrl", "/Admin/Contents/ContentItems");
+
```

Comment:
Tried adding it by injecting IUrlHelper in that class but it's not working so I've set it hardcoded for now for testing purposes.